salt/modules/virt.py

```diff
@@ -1179,23 +1214,73 @@ def _gen_vol_xml(
     return template.render(**context)
 
 
-def _gen_net_xml(name, bridge, forward, vport, tag=None, ip_configs=None):
+def _gen_net_xml(
+    name,
+    bridge,
+    forward,
+    vport,
+    tag=None,
+    ip_configs=None,
+    mtu=None,
+    domain=None,
+    nat=None,
+    interfaces=None,
+    addresses=None,
+    physical_function=None,
+    dns=None,
+):
     """
     Generate the XML string to define a libvirt network
     """
+    vport_context = vport
+    if isinstance(vport, str):
+        vport_context = {"type": vport}
```

Comment:
*suggestion*: An if/else is clearer here, since `vport_context` is falling back to `vport`.
```suggestion
    if isinstance(vport, str):
        vport_context = {"type": vport}
    else:
        vport_context = vport
```

Reply:
Definitely better than `vport_context = {"type": vport} if isinstance(vport, str) else vport` :nauseated_face:

Reply:
OK, I'll change to if/else